fundingmanager.go

```diff
@@ -584,21 +661,232 @@ func (f *fundingManager) handleFundingComplete(fmsg *fundingCompleteMsg) {
+
+	completeChan.IsPending = false
+	err := f.wallet.ChannelDB.MarkChannelAsOpen(fundingPoint)
+	if err != nil {
+		fndgLog.Errorf("error setting channel pending flag to false: %v", err)
+		return
+	}
+
+	// Finally, create and officially open the payment channel!
+	// TODO(roasbeef): CreationTime once tx is 'open'
+	channel, err := lnwallet.NewLightningChannel(f.wallet.Signer, f.wallet.ChainIO,
+		f.chainNotifier, completeChan)
+	if err != nil {
+		fndgLog.Errorf("error creating new lightning channel: %v", err)
+		return
+	}
+
+	// Now that the channel is open, we need to notify a number of
+	// parties of this event.
+
+	// First we send the newly opened channel to the source server
+	peer := f.findPeer(completeChan.IdentityPub)
+	if peer == nil {
+		nilPeerMessage := "Unable to find peer. Peer is nil."
+		fndgLog.Errorf(nilPeerMessage)
+		return
+	}
+	newChanDone := make(chan struct{})
+	newChanReq := &newChannelReq{
+		channel: channel,
+		done:    newChanDone,
+	}
+	peer.newChannels <- newChanReq
+
+	<-newChanDone
+
+	// Close the active channel barrier signalling the
+	// readHandler that commitment related modifications to
+	// this channel can now proceed.
+	f.barrierMtx.Lock()
+	fndgLog.Tracef("Closing chan barrier for ChannelPoint(%v)", fundingPoint)
+	close(f.newChanBarriers[*fundingPoint])
+	delete(f.newChanBarriers, *fundingPoint)
+	f.barrierMtx.Unlock()
+
+	// Afterwards we send the breach arbiter the new channel so it
+	// can watch for attempts to breach the channel's contract by
+	// the remote party.
+	f.breachArbiter.newContracts <- channel
+
+	// With the block height and the transaction index known, we
+	// can construct the compact chainID which is used on the
+	// network to unique identify channels.
+	chanID := lnwire.ChannelID{
+		BlockHeight: confDetails.BlockHeight,
+		TxIndex:     confDetails.TxIndex,
+		TxPosition:  uint16(fundingPoint.Index),
+	}
+
+	// When the funding transaction has been confirmed, the FundingLocked
+	// message is sent to the remote peer so that the existence of the
+	// channel can be announced to the network.
+	fundingLockedMsg := lnwire.NewFundingLocked(fundingPoint, chanID,
+		f.fakeProof.bitcoinSig, f.fakeProof.bitcoinSig, f.idPubKey)
+	f.sendMessages(completeChan.IdentityPub, fundingLockedMsg)
+
+	close(doneChan)
+	return
+}
+
+func (f *fundingManager) processFundingLocked(msg *lnwire.FundingLocked,
+	peerAddress *lnwire.NetAddress) {
+	f.fundingMsgs <- &fundingLockedMsg{msg, peerAddress}
+}
+
+func (f *fundingManager) handleFundingLocked(fmsg *fundingLockedMsg) {
+
+	fundingPoint := *fmsg.msg.ChannelOutpoint
+	channel, err := f.rpcServer.fetchActiveChannel(fundingPoint)
+	if err != nil {
+		fndgLog.Errorf("error looking up channel for outpoint: %v", fundingPoint)
+		return
+	}
+
+	remoteProof := &channelProof{
+		nodeSig:    fmsg.msg.AnnouncementNodeSignature,
+		bitcoinSig: fmsg.msg.AnnouncementBitcoinSignature,
+	}
+
+	// Register the new link with the L3 routing manager so this
```

Comment:
The channel doesn't need to be re-announced to the `ChannelRouter`. The remote peer will broadcast the new advertisements on its own schedule which will eventually be handled by the `ChannelRouter`.